samples_project/Assets/SampleViewer/Samples/Measure/Measure.cs

```diff
@@ -14,272 +14,223 @@
-
-    }
-
-    public static double ConvertUnits(double units, UnitType from, UnitType to)
-    {
-        double[][] factor =
-        {
-            new double[] { 1, 0.001, 0.000621371, 3.28084 },
-            new double[] { 1000,   1,     0.621371,   3280.84},
-            new double[] { 1609.344,     1.609344,       1,   5280},
-            new double[] { 0.3048,    0.0003048,  0.00018939,    1}
-        };
-
-        return units * factor[(int)from][(int)to];
-    }
+	public GameObject Line;
+	public Text GeodeticDistanceText;
+	private String unitTxt;
+	public GameObject LineMarker;
+	public GameObject InterpolationMarker;
+	public float InterpolationInterval = 100;
+	public Dropdown UnitDropdown;
+	public Button ClearButton;
+	private HPRoot hpRoot;
+	private ArcGISMapComponent arcGISMapComponent;
+	private List<GameObject> featurePoints = new List<GameObject>();
+	private Stack<GameObject> stops = new Stack<GameObject>();
+	private double3 lastRootPosition;
+	private double geodeticDistance = 0;
+	private LineRenderer lineRenderer;
+	private ArcGISLinearUnit currentUnit = new ArcGISLinearUnit(ArcGISLinearUnitId.Meters);
+
+	void Start()
+	{
+		// We need HPRoot for the HitToGeoPosition Method
+		hpRoot = FindObjectOfType<HPRoot>();
+
+		// We need this ArcGISMapComponent for the FromCartesianPosition Method
+		// defined on the ArcGISMapComponent.View
+		arcGISMapComponent = FindObjectOfType<ArcGISMapComponent>();
+
+		lineRenderer = Line.GetComponent<LineRenderer>();
+		lastRootPosition = arcGISMapComponent.GetComponent<HPRoot>().RootUniversePosition;
+		unitTxt = " m";
+		UnitDropdown.onValueChanged.AddListener(delegate
+		{
+			UnitChanged();
+		});
+		ClearButton.onClick.AddListener(delegate
+		{
+			ClearLine();
+		});
+
+
+	}
+
+	void Update()
+	{
+
+		if (Input.GetKey(KeyCode.LeftShift) && Input.GetMouseButtonDown(0))
+		{
+			RaycastHit hit;
+			Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+
+			if (Physics.Raycast(ray, out hit))
+			{
+				var lineMarker = Instantiate(LineMarker, hit.point, Quaternion.identity, arcGISMapComponent.transform);
+				var thisPoint = arcGISMapComponent.EngineToGeographic(hit.point);
+
+				lineMarker.GetComponent<ArcGISLocationComponent>().enabled = true;
+				lineMarker.GetComponent<ArcGISLocationComponent>().Position = thisPoint;
+				lineMarker.GetComponent<ArcGISLocationComponent>().Rotation = new ArcGISRotation(0, 90, 0);
+
+				if (stops.Count > 0)
+				{
+					GameObject lastStop = stops.Peek();
+					var lastPoint = lastStop.GetComponent<ArcGISLocationComponent>().Position;
+
+					//calculate distance from last point to this point
+					geodeticDistance += ArcGISGeometryEngine.DistanceGeodetic(lastPoint, thisPoint, new ArcGISLinearUnit(ArcGISLinearUnitId.Meters), new ArcGISAngularUnit(ArcGISAngularUnitId.Degrees), ArcGISGeodeticCurveType.Geodesic).Distance;
+					GeodeticDistanceText.text = "Distance: " + Math.Round(geodeticDistance, 3).ToString() + unitTxt;
```

Comment:
This code does not respect the unit chosen in the UI. Adding a new point will show the distance in meters even when other units are currently selected.

Reply:
nice catch!